addons/mail/static/src/core/common/record.js

```diff
@@ -435,31 +392,66 @@ export class RecordList extends Array {
 }
 
 /**
- * @typedef {Object} RelationDefinition
- * @property {string} [targetModel]
- * @property {Function} [compute]
- * @property {string} [inverse]
- * @property {Function} [onAdd]
- * @property {Function} [onDelete]
+ * @typedef {Object} FieldDefinition
+ * @property {boolean} [ATTR_SYM] true when this is an attribute, i.e. a non-relational field.
+ * @property {boolean} [MANY_SYM] true when this is a many relation.
+ * @property {boolean} [ONE_SYM] true when this is a one relation.
+ * @property {any} [default] the default value of this attribute.
+ * @property {any} [html] whether the attribute is an html field. Useful to automatically markup
+ *   when the insert is trusted.
+ * @property {string} [targetModel] model name of records contained in this relational field.
+ * @property {Function} [compute] if set the field is computed based on provided function.
+ *   The `this` of function is the record, and the function is recalled whenever any field
+ *   in models used by this compute function is changed.
+ * @property {string} [inverse] name of inverse relational field in targetModel.
+ * @property {Function} [onAdd] hook that is called when relation is updated
+ *   with a record being added. Callback param is record being added into relation.
+ * @property {Function} [onDelete] hook that is called when relation is updated
+ *   with a record being deleted. Callback param is record being deleted from relation.
  */
 
 export class Record {
+    /** @param {FieldDefinition} */
+    static isAttr(definition) {
+        return Boolean(definition[ATTR_SYM]);
+    }
+    /**
+     * Determines whether the inserts are considered trusted or not.
+     * Useful to auto-markup html fields when this is set
+     */
+    static trusted = false;
+    /**
+     * Class of markup, useful to detect content that is markup and to
+     * automatically markup field during trusted insert
+     */
+    static Markup = markup("").constructor;
```

Comment:
There is no need to store that on Record class, it can just be a global on the file.